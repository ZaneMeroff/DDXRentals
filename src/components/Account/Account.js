import React, { Component } from 'react';
import './Account.scss'
import SmallListingCard from '../SmallListingCard/SmallListingCard.js';
import one from '../../data/one.png';
import two from '../../data/two.png';
import three from '../../data/three.png';
import four from '../../data/four.png';
import five from '../../data/fivr.png';
import six from '../../data/six.png';
import seven from '../../data/seven.png';
let avatarImg;

class Account extends Component {
  constructor(props) {
    super(props);
    this.state = {
      account: ''
    }
  }

  componentDidMount = () => {
    this.setState({ account: this.props.userData })
    let random = Math.floor(Math.random() * 7);
    let characters = [one, two, three, four, five, six, seven];
    avatarImg = characters[random];
  }

  buildListingCards = () => {
    return this.props.favorites.map(place => {
      return <SmallListingCard
        id={place.id}
        area={place.area}
        name={place.name}
        address={place.address}
        addressZip={place.addressZip}
        cost={place.cost}
        features={place.features}
        beds={place.beds}
        baths={place.baths}
        addFavorite={this.props.removeFavorite}
        />
    })
  }

  render() {
    return (
      <section id='account-page'>
        <div id='account-img'>
          <img src={ avatarImg } id='account-avatar' alt='avatar image from the ssx tricky game' />
          <div id='account-display'>
            <h1 id='account-name'>{this.state.account.name}</h1>
            <h3 id='account-email'>{this.state.account.email}</h3>
            <h3 id='account-value'>Your Custom {this.state.account.value} Ride Summary</h3>
            <p id='account-p'>Welcome back to your Tubular Home Page. Here you can see all your Favorited Hip Hop Hot Spots by scrolling below. Funky Dope! Have fun shredder, and don't forget, Safety First!</p>
          </div>
        </div>
        {this.props.favorites.length === 0 ? <h3 id='no-favs-message'>Welcome rider! There are no favorites here.</h3> : ''}
        <div id='account-listing'>
          <h4>{this.buildListingCards()}</h4>
        </div>
      </section>
    )
  }
}

export default Account;
