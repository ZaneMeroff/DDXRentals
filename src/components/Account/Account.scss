#account-page {
  background-color: #202020;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  padding-top: 100px;
}

#account-img {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

#account-avatar {
  border-radius: 500px 0px 0px 500px;
  border: solid 10px #FFF;
  height: 300px;
}

#account-name {
  color: #202020;
  font-size: 3.5em;
}

#account-email {
  color: #DB49CB;
  font-size: 2em;
}

#account-display {
  display: flex;
  flex-direction: column;
  background-color: rgba(255, 255, 255, 0.8);
  border-radius: 0px 500px 500px 0px;
  padding: 15px;
  width: 50%;
  height: 300px;
  margin-left: 20px;
  border: solid 10px #02A1FF;
  justify-content: space-around;
}

#account-listing {
  display: block;
  overflow-y: scroll;
  padding-bottom: 100px;
  width: 100%;
  margin-left: 50%;
}

#account-p {
  width: 90%;
}
