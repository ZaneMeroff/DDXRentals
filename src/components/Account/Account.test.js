import React from 'react';
import ReactDOM from 'react-dom';
import Account from './Account';
import { shallow } from 'enzyme';

describe('Account', () => {
  let mockUserInfo, mockFavorites;

  beforeEach(() => {
    mockUserInfo = { email: 'ipitythefool@aol.com', name: 'Mr. T', value: 'Business' };
    mockFavorites = {
      about: "RiNo is a burgeoning area with new bars, restaurants and event spaces popping up all the time. Explore this thriving area of Denver today!",
      id: 590,
      listings: ["/api/v1/listings/3", "/api/v1/listings/44", "/apiâ€¦"],
      location: "North of Downtown Denver",
      name: "River North",
      nickName: "RiNo",
    }
  });

  it('should be an instance of Account component', () => {
    const wrapper = shallow(<Account
        userData = {mockUserInfo}
        favorites = {[mockFavorites]}
      />);
      expect(wrapper).toMatchSnapshot();
  });

});
