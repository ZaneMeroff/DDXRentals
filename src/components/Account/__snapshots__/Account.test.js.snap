// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should be an instance of Account component 1`] = `
<section
  id="account-page"
>
  <div
    id="account-img"
  >
    <img
      alt="avatar from the ssx tricky game"
      id="account-avatar"
    />
    <div
      id="account-display"
    >
      <h1
        id="account-name"
      >
        Mr. T
      </h1>
      <h3
        id="account-email"
      >
        ipitythefool@aol.com
      </h3>
      <h3
        id="account-value"
      >
        Your Custom 
        Business
         Ride Summary
      </h3>
      <p
        id="account-p"
      >
        Welcome back to your Tubular Home Page. Here you can see all your Favorited Hip Hop Hot Spots by scrolling below. Funky Dope! Have fun shredder, and don't forget, Safety First!
      </p>
    </div>
  </div>
  <div
    id="account-listing"
  >
    <div>
      <SmallListingCard
        id={590}
        name="River North"
      />
    </div>
  </div>
</section>
`;
